### PR TITLE
deps(gymnasium): update gymnasium version to 0.28.1

### PR DESCRIPTION
**Pull request by muchvo**
## Description

1. Update gymnasium version to 0.28.1

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/OmniSafeAI/safety-gymnasium/issues) for new features and bug fixes) #39 

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/safety-gymnasium/blob/main/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Review by zmsn-2077 — DISMISSED** [review]
LGTM.

**Comment by muchvo** [comment]
> It's time to bump the version to 0.3.0 after the PR #16 finishes.

got it

File: README.md [hunk 1]
```diff
@@ -379,6 +386,15 @@ cd Safety-Gymnasium
 pip install -e .
 ```
 
+### Important Notes
+- If you failed to render on your server, try below:
+  ```bash
+  echo "export MUJOCO_GL=osmesa" >> ~/.bashrc.
+  source ~/.bashrc.
+  apt-get install libosmesa6-dev
+  apt-get install python3-opengl
```
Comment (Gaiejj) [inline README.md:395]: Please note that this method is not work on headless server

**Review by Gaiejj — CHANGES_REQUESTED** [review]

**Review by Gaiejj — APPROVED** [review]
LGTM.

**Review by zmsn-2077 — APPROVED** [review]

**Pull request merged**